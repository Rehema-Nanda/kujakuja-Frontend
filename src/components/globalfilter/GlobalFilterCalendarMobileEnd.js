
import React from 'react';

import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import moment from 'moment';
import GlobalFilterCalendarContainer from "./GlobalFilterCalendarContainer";
import {CalendarConfig} from "./CalendarConfig";

export default class GlobalFilterCalendarMobileEnd extends GlobalFilterCalendarContainer {

    constructor(props) {
        super(props);
        this.handleDateChangeEnd = this.handleDateChangeEnd.bind(this);
        this.state = {
            selectedDateRangeType: this.getDateRangeType(),
            isMobileView: true
        };
    }

    handleDateChangeEnd(endDate) {
        this.handleDateRangeSelect(this.props.dateStart, moment(endDate), CalendarConfig.IS_END_DATE);
        this.props.displayMobileFilter();
    }

    render() {

        return (

            <div className="date-pickers-holder fadein">
                <DatePicker inline
                    className="date-picker-startdate"
                    withPortal={this.props.withPortal}
                    selected={this.props.dateEnd.toDate()}
                    selectsEnd
                    ref={CalendarConfig.END_DATE_PICKER_REF}
                    startDate={this.props.dateStart.toDate()}
                    endDate={this.props.dateEnd.toDate()}
                    maxDate={moment.utc().startOf('day').toDate()}
                    calendarContainer={this.calendarContainer.bind(this)}
                    onChange={this.handleDateChangeEnd} />
            </div>

        );
    }

}
