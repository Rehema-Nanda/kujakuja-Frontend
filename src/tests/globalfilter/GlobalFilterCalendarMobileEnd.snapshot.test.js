import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import GlobalFilterCalendarMobileEnd from '../../components/globalfilter/GlobalFilterCalendarMobileEnd';

const testProps = {
    dateStart: moment.utc('2017-01-01'),
    dateEnd: moment.utc('2018-01-01'),
    setDateStartEnd: jest.fn(),
    displayMobileFilter: jest.fn()
};

const wrapper = renderer.create(<GlobalFilterCalendarMobileEnd {...testProps} />).toJSON();

it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
});
