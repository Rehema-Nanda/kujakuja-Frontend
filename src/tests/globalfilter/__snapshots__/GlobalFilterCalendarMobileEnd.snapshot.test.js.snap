// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="date-pickers-holder fadein"
>
  <div>
    <div>
      <div
        className="react-datepicker"
      >
        <div
          className="react-datepicker__triangle"
        />
        <div
          className="custom-date-parent"
        >
          <button
            className="btn btn-light custom-date "
            onClick={[Function]}
            type="button"
          />
          <button
            className="btn btn-light custom-date "
            onClick={[Function]}
            type="button"
          />
          <button
            className="btn btn-light custom-date "
            onClick={[Function]}
            type="button"
          />
          <button
            className="btn btn-light custom-date selected btn-dark "
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <button
            className="react-datepicker__navigation react-datepicker__navigation--previous"
            onClick={[Function]}
            type="button"
          >
            Previous Month
          </button>
          <button
            className="react-datepicker__navigation react-datepicker__navigation--next"
            onClick={[Function]}
            type="button"
          >
            Next Month
          </button>
          <div
            className="react-datepicker__month-container"
          >
            <div
              className="react-datepicker__header"
            >
              <div
                className="react-datepicker__current-month"
              >
                December 2017
              </div>
              <div
                className="react-datepicker__header__dropdown react-datepicker__header__dropdown--scroll"
                onFocus={[Function]}
              />
              <div
                className="react-datepicker__day-names"
              >
                <div
                  className="react-datepicker__day-name"
                >
                  Su
                </div>
                <div
                  className="react-datepicker__day-name"
                >
                  Mo
                </div>
                <div
                  className="react-datepicker__day-name"
                >
                  Tu
                </div>
                <div
                  className="react-datepicker__day-name"
                >
                  We
                </div>
                <div
                  className="react-datepicker__day-name"
                >
                  Th
                </div>
                <div
                  className="react-datepicker__day-name"
                >
                  Fr
                </div>
                <div
                  className="react-datepicker__day-name"
                >
                  Sa
                </div>
              </div>
            </div>
            <div
              aria-label="month-2017-12"
              className="react-datepicker__month"
              onMouseLeave={[Function]}
              role="listbox"
            >
              <div
                className="react-datepicker__week"
              >
                <div
                  aria-disabled={false}
                  aria-label="day-26"
                  className="react-datepicker__day react-datepicker__day--026 react-datepicker__day--in-range react-datepicker__day--weekend react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  26
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-27"
                  className="react-datepicker__day react-datepicker__day--027 react-datepicker__day--in-range react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  27
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-28"
                  className="react-datepicker__day react-datepicker__day--028 react-datepicker__day--in-range react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  28
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-29"
                  className="react-datepicker__day react-datepicker__day--029 react-datepicker__day--in-range react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  29
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-30"
                  className="react-datepicker__day react-datepicker__day--030 react-datepicker__day--in-range react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  30
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-1"
                  className="react-datepicker__day react-datepicker__day--001 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  1
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-2"
                  className="react-datepicker__day react-datepicker__day--002 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  2
                </div>
              </div>
              <div
                className="react-datepicker__week"
              >
                <div
                  aria-disabled={false}
                  aria-label="day-3"
                  className="react-datepicker__day react-datepicker__day--003 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  3
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-4"
                  className="react-datepicker__day react-datepicker__day--004 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  4
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-5"
                  className="react-datepicker__day react-datepicker__day--005 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  5
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-6"
                  className="react-datepicker__day react-datepicker__day--006 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  6
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-7"
                  className="react-datepicker__day react-datepicker__day--007 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  7
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-8"
                  className="react-datepicker__day react-datepicker__day--008 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  8
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-9"
                  className="react-datepicker__day react-datepicker__day--009 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  9
                </div>
              </div>
              <div
                className="react-datepicker__week"
              >
                <div
                  aria-disabled={false}
                  aria-label="day-10"
                  className="react-datepicker__day react-datepicker__day--010 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  10
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-11"
                  className="react-datepicker__day react-datepicker__day--011 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  11
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-12"
                  className="react-datepicker__day react-datepicker__day--012 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  12
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-13"
                  className="react-datepicker__day react-datepicker__day--013 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  13
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-14"
                  className="react-datepicker__day react-datepicker__day--014 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  14
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-15"
                  className="react-datepicker__day react-datepicker__day--015 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  15
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-16"
                  className="react-datepicker__day react-datepicker__day--016 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  16
                </div>
              </div>
              <div
                className="react-datepicker__week"
              >
                <div
                  aria-disabled={false}
                  aria-label="day-17"
                  className="react-datepicker__day react-datepicker__day--017 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  17
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-18"
                  className="react-datepicker__day react-datepicker__day--018 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  18
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-19"
                  className="react-datepicker__day react-datepicker__day--019 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  19
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-20"
                  className="react-datepicker__day react-datepicker__day--020 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  20
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-21"
                  className="react-datepicker__day react-datepicker__day--021 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  21
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-22"
                  className="react-datepicker__day react-datepicker__day--022 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  22
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-23"
                  className="react-datepicker__day react-datepicker__day--023 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  23
                </div>
              </div>
              <div
                className="react-datepicker__week"
              >
                <div
                  aria-disabled={false}
                  aria-label="day-24"
                  className="react-datepicker__day react-datepicker__day--024 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  24
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-25"
                  className="react-datepicker__day react-datepicker__day--025 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  25
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-26"
                  className="react-datepicker__day react-datepicker__day--026 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  26
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-27"
                  className="react-datepicker__day react-datepicker__day--027 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  27
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-28"
                  className="react-datepicker__day react-datepicker__day--028 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  28
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-29"
                  className="react-datepicker__day react-datepicker__day--029 react-datepicker__day--in-range"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  29
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-30"
                  className="react-datepicker__day react-datepicker__day--030 react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  30
                </div>
              </div>
              <div
                className="react-datepicker__week"
              >
                <div
                  aria-disabled={false}
                  aria-label="day-31"
                  className="react-datepicker__day react-datepicker__day--031 react-datepicker__day--selected react-datepicker__day--range-end react-datepicker__day--in-range react-datepicker__day--weekend"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  31
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-1"
                  className="react-datepicker__day react-datepicker__day--001 react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  1
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-2"
                  className="react-datepicker__day react-datepicker__day--002 react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  2
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-3"
                  className="react-datepicker__day react-datepicker__day--003 react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  3
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-4"
                  className="react-datepicker__day react-datepicker__day--004 react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  4
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-5"
                  className="react-datepicker__day react-datepicker__day--005 react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  5
                </div>
                <div
                  aria-disabled={false}
                  aria-label="day-6"
                  className="react-datepicker__day react-datepicker__day--006 react-datepicker__day--weekend react-datepicker__day--outside-month"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="option"
                >
                  6
                </div>
              </div>
            </div>
          </div>
        </div>
        <h5
          className="choose-from-end-date"
          style={
            Object {
              "display": "none",
            }
          }
        >
          undefined
        </h5>
        <h5
          className="choose-from-end-date"
          style={
            Object {
              "display": "none",
            }
          }
        >
          undefined
        </h5>
        <h5
          className="choose-from-end-date"
          style={
            Object {
              "display": "none",
            }
          }
        >
          undefined
        </h5>
        <h5
          className="choose-from-end-date"
          style={
            Object {
              "display": "none",
            }
          }
        >
          undefined
        </h5>
        <h5
          className="choose-from-end-date"
          style={
            Object {
              "display": "none",
            }
          }
        >
          undefined
        </h5>
      </div>
    </div>
  </div>
</div>
`;
